clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -914,7 +915,7 @@ public void flush() {
     @Override
     public List<PartitionInfo> partitionsFor(String topic) {
         try {
-            return waitOnMetadata(topic, null, maxBlockTimeMs).cluster.partitionsForTopic(topic);
+            return waitOnMetadata(Utils.notNull(topic), null, maxBlockTimeMs).cluster.partitionsForTopic(topic);
```

Comment:
`Metadata.add` should also not allow a null `topic` either. Also, we provide an error message if a user provides a null `topic` here.

Reply:
updated the PR with changes. pls check